### PR TITLE
fix(android/engine): Support for U_xxxx_yyyy subkeys

### PR DESCRIPTION
**Pull request by darcywong00**
Follow-on to #5894 

This updates the keycap text in the Android subkeys (longpress keys) to handle U_xxxx_yyyy keys.

Additional updates to the u_xxxx_yyyy_2858 test keyboard
* Assign English language so kmp will install in mobile devices
* Include noto font for the Old Italic SMP characters
* Add some longpress keys

## User Testing
Perform the following tests in an Android emulator with:
    ~* Android SDK 21 (Note: currently the SMP keys are showing tofu on both the longpress keycaps and output)~ Only test for SDK 29 since the tofu issue will be resolved in #5914
    * Android SDK 29
Ignore the blank suggestion bar. The test keyboard arbitrarily is assigned "English" even though the special characters are Amharic and Old Italic script.

* TEST_U_xxxx_yyyy : Tests U_xxxx_yyyy is handled on keys and longpress keys
1. Load the Keyman app
2. Install the [u_xxxx_yyyy_2858](https://darcywong00.github.io/examples/u2858/u_xxxx_yyyy_2858.kmp) keyboard 
3. For the keys in the screenshot:
    1. Press the key
    2. Verify the added string matches the keycap
    3. If available, press the longpress key
    4. Verify the added string matches the longpress keycap
Screenshot_1637030471

### CODE REVIEW

**Comment by darcywong00** [comment]
>  Android SDK 21 (Note: currently the SMP keys are showing tofu on both the longpress keycaps and output)

Per @mcdurdin - I'll split this into a separate issue

**Comment by MakaraSok** [comment]
* **TEST_U_xxxx_yyyy: PASSED** The strings added do match the keycaps.

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java [hunk 1]
```diff
@@ -676,18 +676,30 @@ private void sendKMWError(int lineNumber, String sourceId, String message) {
...
+        // Split U_xxxx_yyyy
+        String[] codePoints = values[j].split("_");
+        for (String codePoint : codePoints) {
+          int codePointValue = Integer.parseInt(codePoint, 16);
+          if (((0x0 <= codePointValue) && (codePointValue <= 0x1F)) || ((0x80 <= codePointValue) && (codePointValue <= 0x9F))) {
+            continue;
+          } else {
+            // TODO: \\uxxxxxx will need to be handled with title.codePointAt(c)
+            title += new String(Character.toChars(codePointValue));
+          }
```
Comment (mcdurdin) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java:701]: We need to check that the value is `<=0x10FFFF` as well. Also, should be able to remove innermost sets of parentheses which will make it easier to read.

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java [hunk 2]
```diff
@@ -676,18 +676,30 @@ private void sendKMWError(int lineNumber, String sourceId, String message) {
...
-        // TODO: \\uxxxxxx will need to be handled with title.codePointAt(c)
-        title += String.valueOf((char) c);
+        // Split U_xxxx_yyyy
+        String[] codePoints = values[j].split("_");
+        for (String codePoint : codePoints) {
+          int codePointValue = Integer.parseInt(codePoint, 16);
+          if (((0x0 <= codePointValue) && (codePointValue <= 0x1F)) || ((0x80 <= codePointValue) && (codePointValue <= 0x9F))) {
+            continue;
+          } else {
+            // TODO: \\uxxxxxx will need to be handled with title.codePointAt(c)
```
Comment (mcdurdin) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java:699]: What is this TODO for?

**Review by mcdurdin — CHANGES_REQUESTED** [review]

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java (hunk 2 above)
Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java:699]: I added that in #357 but forget why. I'll remove it.

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java (hunk 1 above)
Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java:701]: Does this also apply to 
https://github.com/keymanapp/keyman/blob/649dd331c0a9f087948e96a85643515a3023b1bf/web/source/osk/oskKey.ts#L418-L424  ?

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.151-alpha](https://keyman.com/downloads/releases/alpha/15.0.151)

**Pull request merged**